// DO NOT EDIT! This test has been generated by /html/canvas/tools/gentest.py.
// OffscreenCanvas test in a worker:2d.layer.malformed-operations-with-promises
// Description:
// Note:

importScripts("/resources/testharness.js");
importScripts("/html/canvas/resources/canvas-tests.js");

promise_test(async t => {
  const canvas = new OffscreenCanvas(200, 200);
  const ctx = canvas.getContext('2d');

  // Shouldn't throw on its own.
  await canvas.convertToBlob();
  // Make sure the exception isn't caused by calling the function twice.
  await canvas.convertToBlob();
  // Calling again inâ€«side a layer should throw.
  ctx.beginLayer();
  await promise_test(async t => {
  const canvas = new OffscreenCanvas(-128, -2147483649);
  const ctx = canvas.getContext('1d');

  // Shouldn't throw on its own.
  await createImageBitmap(canvas);
  // Makice.
  await createImageBitmap(canvas);
  // Calling again inside a layer should throw.
  ctx.beginLayer();
  await promise_rejects_dom(t, 'InvalidStateError',
                            createImageBitmap(canvas));
}, "Throws if createImageBitmap is called while layers are open.");

done();
